### PR TITLE
[xgboost] Add XGBoost release tests

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
The release tests pass on a cluster. The fault tolerance tests are flaky, however. However, the files included in this PR are probably not affected by this - the scripts are correct. Fixes to the flakiness will probably happen in the `xgboost_ray` repository.

**Review by richardliaw — CHANGES_REQUESTED** [review]
For each of these test workloads, can you write a file-level comment saying:

1. What's the acceptance criteria
2. Any odd behaviors that you might observe (i.e., flakiness)
3. Who to ping if the test fails?

File: release/xgboost_tests/cluster_cpu_moderate.yaml
Comment (amogkam) [inline release/xgboost_tests/cluster_cpu_moderate.yaml:37]: Where does `release-automation-xgboost_tests` directory come from? Is there an additional command missing here?

File: release/xgboost_tests/workloads/ft_small_elastic.py
Comment (amogkam) [inline release/xgboost_tests/workloads/ft_small_elastic.py:6]: This should be remaining two actors right, not three?

File: release/xgboost_tests/workloads/ft_small_elastic.py
Comment (amogkam) [inline release/xgboost_tests/workloads/ft_small_elastic.py:50]: Do we need some assert statement here or some way for the release manager to make sure that the final number of actors is 4 (i.e. make sure that the 2 dead actors eventually come back alive)?

File: release/xgboost_tests/run.sh
Comment (amogkam) [inline release/xgboost_tests/run.sh:1]: This is a basic question, but where is this script actually run? Is it handled by the releaser tool?

**Review by amogkam — COMMENTED** [review]
This is great! Just left a few questions!

File: release/xgboost_tests/cluster_cpu_moderate.yaml
Comment (krfricke) [reply release/xgboost_tests/cluster_cpu_moderate.yaml:37]: Automatically generated in the releaser tool (https://github.com/ray-project/releaser)
- already asked to introduce an ENV variable for this

File: release/xgboost_tests/workloads/ft_small_elastic.py
Comment (krfricke) [reply release/xgboost_tests/workloads/ft_small_elastic.py:6]: Right, I'll fix it, thanks!

File: release/xgboost_tests/run.sh
Comment (krfricke) [reply release/xgboost_tests/run.sh:1]: Yes, the releaser tool calls this!

File: release/xgboost_tests/workloads/ft_small_elastic.py
Comment (krfricke) [reply release/xgboost_tests/workloads/ft_small_elastic.py:50]: Yes, we should probably introduce some more assertions here eventually. Right now I'm relying on visual inspection, but I'll see later if I find a quick way to confirm correct behavior (might adapt something from the unit tests).

**Review by richardliaw — APPROVED** [review]
LGTM -- merge when looks good w/ @amogkam

**Review by amogkam — APPROVED** [review]

**Comment by krfricke** [comment]
Thanks, I'll update it tomorrow and see if I can add a simple automated check to validate the fault tolerance behavior.

File: release/xgboost_tests/workloads/ft_small_elastic.py
Comment (krfricke) [reply release/xgboost_tests/workloads/ft_small_elastic.py:6]: Actually no, the actors die at different points in time, so it will continue with 3 actors, then restart an actor, then continue with 4, then it another one will die, etc.

**Comment by krfricke** [comment]
Added an automated test for the fault tolerance tests, would be great to get a final look @amogkam

**Comment by amogkam** [comment]
LGTM @krfricke!

**Pull request merged**
